detox/src/ipc/ipc.test.js

```diff
@@ -116,69 +186,108 @@ describe('IPC', () => {
         await ipcClient2.init();
       });
 
-      it('should report test results', async () => {
-        const testResults1 = [
-          {
+      describe('reportTestResults', () => {
+        it('should report test results and synchronize test results', async () => {
+          const testResults = [
+            {
+              testFilePath: 'file1',
+              success: true,
+            },
+          ];
+
+          await ipcClient1.reportTestResults(testResults);
+          expect(ipcServer.sessionState.testResults).toEqual(testResults);
+          expect(ipcClient1.sessionState.testResults).toEqual(testResults);
+
+          await sleep(10); // broadcasting happens with a delay
+          expect(ipcClient2.sessionState.testResults).toEqual(testResults);
+        });
+
+        it('should update test results with tests from another clients', async () => {
+          const testResults1 = [{
             testFilePath: 'file1',
             success: true,
-          },
-          {
-            testFilePath: 'file2',
-            success: false,
-            testExecError: { name: 'baz', message: 'qux', stack: 'quux' },
-            isPermanentFailure: true,
-          },
-        ];
-
-        await ipcClient1.reportTestResults(testResults1);
-
-        let expected = testResults1;
-        expect(ipcServer.sessionState.testResults).toEqual(expected);
-        expect(ipcClient1.sessionState.testResults).toEqual(expected);
-        expect(ipcClient2.sessionState.testResults).toEqual(expected);
-
-        const testResults2 = [
-          {
-            testFilePath: 'file3',
-            success: true,
-          },
-        ];
-
-        await ipcClient2.reportTestResults(testResults2);
+          }];
+
+          const testResults2 = [
+            {
+              testFilePath: 'file2',
+              success: false,
+              testExecError: { name: 'baz', message: 'qux', stack: 'quux' },
+              isPermanentFailure: false,
+            },
+          ];
+
+          await ipcClient1.reportTestResults(testResults1);
+          await ipcClient2.reportTestResults(testResults2);
+          await sleep(10); // broadcasting might happen with a delay
+
+          const expected = [...testResults2, ...testResults1];
+          expect(ipcServer.sessionState.testResults).toEqual(expected);
+          expect(ipcClient1.sessionState.testResults).toEqual(expected);
+          expect(ipcClient2.sessionState.testResults).toEqual(expected);
+        });
 
-        expected = [...testResults2, ...testResults1];
-        expect(ipcServer.sessionState.testResults).toEqual(expected);
-        expect(ipcClient1.sessionState.testResults).toEqual(expected);
-        expect(ipcClient2.sessionState.testResults).toEqual(expected);
+        it('should update test results with retried tests', async () => {
```

Comment:
Added a case when an IPC client sends data about the same test, e.g. when retrying to run a failed test file.

Reply:
🙌🏼